pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/NamespaceResources.java

```diff
@@ -381,13 +383,13 @@ public CompletableFuture<Void> runWithMarkDeleteAsync(TopicName topic,
     // clear resource of `/loadbalance/bundle-data/{tenant}/{namespace}/` in metadata-store
     public CompletableFuture<Void> deleteBundleDataAsync(NamespaceName ns) {
         final String namespaceBundlePath = joinPath(BUNDLE_DATA_BASE_PATH, ns.toString());
-        return getStore().deleteRecursive(namespaceBundlePath);
```

Comment:
Just to confirm this logic change,

Previously, we have not deleted any bundle data because we were deleting it in the wrong path here?

Reply:
Not wrong path, but wrong places. There are two metadata store: local store and global zk. 
When deleting a namespace, we will delete znode under the path `/loadbalance/bundle-data` in `local` metadata store instead of `global` metadata store.